pkg/tsdb/cloudwatch/metric_find_query.go

```diff
@@ -775,12 +789,12 @@ func (e *cloudWatchExecutor) getMetricsForCustomMetrics(region, namespace, profi
 
 var dimensionsCacheLock sync.Mutex
 
-func (e *cloudWatchExecutor) getDimensionsForCustomMetrics(region, namespace, profile string) ([]string, error) {
+func (e *cloudWatchExecutor) getDimensionsForCustomMetrics(props props) ([]string, error) {
 	dimensionsCacheLock.Lock()
 	defer dimensionsCacheLock.Unlock()
 
 	bldr := strings.Builder{}
-	for i, s := range []string{profile, region, namespace} {
+	for i, s := range props.toArray() {
```

Comment:
To simplify here, the algorithm for computing the cache key should be extracted instead. But I would do that in another PR than this hotfix.